### ISSUE TITLE
oc-sync client drops locally unchanged folders after they had been taken off sync 

### ISSUE THREAD

**Issue by finius** [owncloud/client#3142]
When you completely sync your oc-root folder with your client and then pause the sync to put one folder out of the sync process, then the oc-client drops the local folder that had been taken out of the sync process when resuming the snycing - but just in case you had made NO changes to that folder before sync resumed!

If you did some changes on some data within that folder after stopping sync and taking the folder out of the sync, but before resuming the snyc process, it just leaves the folder as it is, without any deletions or changes, so you can put it back to sync sometime. (just as expected)

This shouldn't be the normal behaviour. 
There are many people that like to keep their cloud data stored locally, even if it shall not be synced permanently - for example because it is a plenty of files in that folder that takes hours to get explored on the server side by every new sync process.

Why not keeping the local data folder that had been taken off sync, just as the client does with folders, that had been expected changes after being deselected for sync? At least the user should have the choice of keeping the files locally when deselecting the synced folder.

**Comment by moscicki** [comment]
I was explained that this behaviour is by design (and actually has some good side to it). To achieve what you want you may add your folder to the list of ignored patterns ("Edit Ignore Files").

**Comment by finius** [comment]
Ok, that's an idea, but, nevertheless, it feels like a workaround, because anybody around us found it a strange behaviour until we realized that the difference of erasing a folder and keeping it, was the change of Content within. 
You may overhink the user experience cause IMHO this is unfortunately non intuitivly design. At least some Messages should be displayed to inform common users concerning this.

**Comment by MTRichards** [comment]
So I understand, let me write what I think you have asked for here.

When you use selective sync to stop syncing a directory, you would like the data to remain on the local drive. 

Is that correct?

**Comment by finius** [comment]
That's correct!

You may have a directory with a huge amount of files (i.e. >200.000) and folders that take almost an hour to get crawled by the client before any data is synced - even if the data in it nearly never changes (i.e. a directory with software installation files or some kind of big archive folder hierarchy). 
Nevertheless, sometimes you have to use this documents offline, so you Need it locally.
And, from time to time it could be reasonable to put this folder back to sync (i.e. you added a licence file or you create a new folder within  your archive).

**Comment by alexpyattaev** [comment]
Your solution is to use another sync tree - then you can disable the whole tree, instead of selected folders. But if ANY files are deleted locally that are fine on the server it is a bug.

**Comment by finius** [comment]
That would be my only reasonable workaround. But I am not able to create two independant sync processes using...
1) ...the same ownCloud-user
2) ...with different folder selections out of the users data dir

Let's say the user got 4 folders in the root dir of his ownCloud: I.e. A, B, C and D. A and B are smaller folders with regulary changed content that always has immediately to be synced. C and D are both heavy folders with a huge file and subfolder hierarchy in it, but the content just changes rarely, so a manual sync every now and then is sufficient.

If I'm now setting up a new client for the user (using the latest release for Windows), I can decide to sync the whole users ownCloud directory or just single folders of it. But I am neither able to deselect the root completely (the files in the root directory are synced in any case) nor to ad a second folder sync process through the GUI. If I try so, I get an warning, that I already snyc the root. But, I can't proceed, and - as I told before - I can not completely deselect the root during the creation of the initial sync process.

**Comment by RichardPillay** [comment]
I found this discussion when I searched to see if there was a bug report already existing. I was surprised to find the behaviour is by design. IMO, it was a very unintuitive and very dangerous decision.

Like Finius, I decided to stop syncing several folders under my root. They were all folders with massive collections of files and with some with large files that were preventing successful syncs because the computer would not be running long enough to finish the sync before I turned it off again.

My expectation was that when I unticked the folders in the choose what to sync dialog box, that those folders would no longer be included in future sync operations. I did not want, (or expect) my data to be removed, especially since this was my only copy, as I don't sync to/from multiple computers. One of those folders was an SVN database, and even though I've now got the data back , I'm unsure of whether it's all back.

MTRichards asked:
*When you use selective sync to stop syncing a directory, you would like the data to remain on the local drive.*

Again in my opinion, when a directory is removed from sync **then no further action must be taken by the system.** The user only asked that the folder no longer be synced. We shouldn't assume he want's any further action, because the possibilities are endless, and the possibilities also include no further action.

**Comment by RichardPillay** [comment]
On thinking back, is it possible that this behaviour is at the heart of some of those "Owncloud deleted my data" posts that Google keeps spewing up?

**Comment by gitkrax** [comment]
Found this while looking for an answer why my files were deleted. This seems to be a duplicate of #2502.

**Comment by phil-davis** [comment]
Note that in the multi-account client 2.0 for the issue of having a big folder "B1" that does not change often, and just manually syncing it every now and then, you can:
1) Setup the account as normal, syncing the root from master to somewhere on the client, and choose not to sync "B1". In this account new folders that appear on the server or are created on the client will automagically sync. It will be everything expect folder "B1".
2) Add an account again, put in the same ownCloud server and account details, skip setting up the sync folder. Then use the "Add Folder to Synchronize" button to add just a sync of B! on the server to somewhere on the client device.

Now (1) can do its regular thing and you can pause sync of (2).

Not entirely obvious or easy to realise for a new user, but it should work.

**Comment by trespanes** [comment]
I just stumpled over this today. I love owncloud but this is really **very** unintuitive and I hope it gets fixed soon. I synched a local folder to a remote machine with a 30GB storage limit. After some time some of the subfolders got to be too big and I had locally more than >30GB which prevented my machine from syncing. So I wanted to exclude some of the local folders that got too big from the sync and what actually happened is that they got deleted locally and remain on the remote machine. That does not make any sense for the average user. So no I have to download ~ 25 GB to start working on my projects again (thanks for the time-loss) and then set up a whole new synchronization scheme? Seriously? 

No, I'm not going to implement any of the proposed complex workarounds as the vast majority of non-nerdy users will not do. Explain that to somebody who uses Dropbox or gdrive and good luck convincing them to use owncloud.
From a technical Viewpoint it would be very easy to implement checkboxes for each folder wheter they sould remain (local) (remote) (local/remote) in the client ... so please: implement something like that and let people freely choose how to use the service instead of patronising them through design-choices. That's what Drobox does for profit purposes.

**Comment by ogoffart** [comment]
duplicate of https://github.com/owncloud/client/issues/2502
